tests/integration-tests/conftest.py

```diff
@@ -424,34 +424,46 @@ def cfn_stacks_factory(request):
         factory.delete_all_stacks()
 
 
-# FIXME: we need to find a better solution to this since AZs are independently mapped to names for each AWS account.
-# https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html
 AVAILABILITY_ZONE_OVERRIDES = {
-    # c5.xlarge is not supported in us-east-1e
-    # FSx Lustre file system creation is currently not supported for us-east-1e
-    # m6g.xlarge is not supported in us-east-1c or us-east-1e
-    "us-east-1": ["us-east-1a", "us-east-1b", "us-east-1d", "us-east-1f"],
-    # m6g.xlarge is not supported in us-east-2a
-    "us-east-2": ["us-east-2b", "us-east-2c"],
-    # c4.xlarge is not supported in us-west-2d
-    "us-west-2": ["us-west-2a", "us-west-2b", "us-west-2c"],
-    # c5.xlarge is not supported in ap-southeast-2a
-    "ap-southeast-2": ["ap-southeast-2b", "ap-southeast-2c"],
-    # m6g.xlarge is not supported in ap-northeast-1d
-    "ap-northeast-1": ["ap-northeast-1a", "ap-northeast-1c"],
-    # c4.xlarge is not supported in ap-northeast-2b
-    "ap-northeast-2": ["ap-northeast-2a", "ap-northeast-2c"],
-    # c5.xlarge is not supported in ap-southeast-1c
-    "ap-southeast-1": ["ap-southeast-1a", "ap-southeast-1b"],
-    # c4.xlarge is not supported in ap-south-1c
-    "ap-south-1": ["ap-south-1a", "ap-south-1b"],
-    # NAT Gateway not available in sa-east-1b
-    "sa-east-1": ["sa-east-1a", "sa-east-1c"],
-    # m6g.xlarge instances not available in eu-west-1c
-    "eu-west-1": ["eu-west-1a", "eu-west-1b"],
+    # c5.xlarge is not supported in use1-az3
+    # FSx Lustre file system creation is currently not supported for use1-az3
+    # m6g.xlarge is not supported in use1-az2 or use1-az3
+    "us-east-1": ["use1-az6", "use1-az1", "use1-az4", "use1-az5"],
+    # m6g.xlarge is not supported in use2-az1
+    "us-east-2": ["use2-az2", "use2-az3"],
+    # c4.xlarge is not supported in usw2-az4
+    "us-west-2": ["usw2-az2", "usw2-az1", "usw2-az3"],
+    # c5.xlarge is not supported in apse2-az3
+    "ap-southeast-2": ["apse2-az1", "apse2-az2"],
+    # m6g.xlarge is not supported in apne1-az2
+    "ap-northeast-1": ["apne1-az4", "apne1-az1"],
+    # c4.xlarge is not supported in apne2-az2
+    "ap-northeast-2": ["apne2-az1", "apne2-az3"],
+    # c5.xlarge is not supported in apse1-az3
+    "ap-southeast-1": ["apse1-az2", "apse1-az1"],
+    # c4.xlarge is not supported in aps1-az2
+    "ap-south-1": ["aps1-az1", "aps1-az3"],
+    # NAT Gateway not available in sae1-az2
+    "sa-east-1": ["sae1-az1", "sae1-az3"],
+    # m6g.xlarge instances not available in euw1-az3
+    "eu-west-1": ["euw1-az1", "euw1-az2"],
 }
 
 
+@pytest.fixture(scope="function")
+def random_az_selector(request):
+    """Select random AZs for a given region."""
+
+    def _get_random_availability_zones(region, k=1, default_value=None):
```

Comment:
minor: the name of the parameter "k" is not self-explanatory. 
This requires to check the code or the comment to understand what it means.

It's not a problem for this function but it could help in the caller, e.g.
`random_az_selector(region, k=2, default_value="")`

Reply:
i agree. i made the change.